cg/services/analysis_starter/configurator/abstract_service.py

```diff
@@ -1,8 +1,10 @@
 from abc import ABC
 
+from cg.services.analysis_starter.configurator.abstract_model import CaseConfig
+
 
 class Configurator(ABC):
```

Comment:
Looking at this now I think the configurator can be a wrapper around the detailed implementations of the FileContentCreators perhaps.
Maybe it is worthwhile to evaluate this approach as it might be able to reduce multi-inheritiance of classes which can be cumbersome to go through.

```suggestion
class Configurator:
    def __init__(self, path_creator: analysis_path_manager, config_creator: FileContentCreator, params_creator: FileContentCreator, sample_sheet_creator: FileContentCreator):
        self.path_creator = path_creator
        self.config_creator = config_creator
        self.params_creator = params_creator
        self.sample_sheet_creator = sample_sheet_creator
        # other required things such a store, lims etc.
        # you can give an optional service in case there is something mega special to setup for a pipeline 

    def create_configurations(self, case_id) -> CaseConfig:
        self.config_creator.create(case_id) # let them internally figure out what they need based on the case
        self.params_creator.create(case_id)
        self.sample_sheet_creator(case_id)
        return CaseConfig #however this gets created
```

Reply:
This configurator is also the parent of all the non-Nextflow pipelines, which unfortunately don't follow this pattern. There are no config, sample sheets nor params files in MicroSALT, for example